examples/notebooks/blogposts/unittests_in_beam.ipynb

```diff
@@ -0,0 +1,259 @@
+        "from apache_beam.options.pipeline_options import PipelineOptions\n",
+        "\n",
+        "class CustomClass(beam.DoFn):\n",
+        "  def custom_function(x):\n",
+        "          ...\n",
+        "          # returned_record = requests.get(\"http://my-api-call.com\")\n",
+        "          ...\n",
+        "          # if len(returned_record)!=10:\n",
+        "          # raise ValueError(\"Length of record does not match expected length\")\n",
+        "          return x\n",
+        "\n",
+        "  with beam.Pipeline() as p:\n",
+        "    result = (\n",
+        "            p\n",
+        "            | ReadFromText(\"/content/sample_data/anscombe.json\")\n",
+        "            | beam.ParDo(lambda x: CustomClass.custom_function(x))\n",
+        "            | WriteToText(\"/content/\")\n",
+        "    )"
+      ],
+      "metadata": {
+        "id": "Ktk9EVIFzGfP"
+      },
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "**Example Pipeline 1**\n"
+      ],
+      "metadata": {
+        "id": "IVjBkewt1sLA"
+      }
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "# This function is going to return the square the integer at the first index of our record.\n",
+        "def compute_square(element):\n",
+        "  return int(element[1])**2\n",
+        "\n",
+        "with beam.Pipeline() as p1:\n",
+        "    result = (\n",
+        "        p1\n",
+        "        | ReadFromText(\"/content/sample_data/california_housing_test.csv\",skip_header_lines=1)\n",
+        "        | beam.Map(compute_square)\n",
+        "        | WriteToText(\"/content/\")\n",
+        "    )"
+      ],
+      "metadata": {
+        "id": "oHbSvOUI1pOe"
+      },
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "**Example Pipeline 2**"
+      ],
+      "metadata": {
+        "id": "Mh3nZZ1_12sX"
+      }
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "with beam.Pipeline() as p2:\n",
+        "    result = (\n",
+        "        p2\n",
+        "        | ReadFromText(\"/content/sample_data/anscombe.json\")\n",
+        "        | beam.Map(str.strip)\n",
+        "        | WriteToText(\"/content/sample_data/\")\n",
+        "    )"
+      ],
+      "metadata": {
+        "id": "hmO1Chl51vPG"
+      },
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "**Unit Tests for Pipelines**"
+      ],
+      "metadata": {
+        "id": "uoNJLQl_15gj"
+      }
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "# The following packages are imported for unit testing.\n",
+        "import unittest\n",
+        "import apache_beam as beam\n",
+        "from apache_beam.testing.test_pipeline import TestPipeline\n",
```

Comment:
Is three any advantage to users of using `TestPipeline`?

Reply:
AFAICT, no distinct advantages outside of the reasons mentioned [here](https://github.com/apache/beam/blob/master/sdks/python/apache_beam/testing/test_pipeline.py#L36), as well as it being the de-facto choice for tests from previous documentation.